lib/proxy/conn.c

```diff
@@ -816,9 +816,14 @@ conn_t *proxy_conn_get_server_conn(pool *p, struct proxy_session *proxy_sess,
   }
 
   server_conn = pr_inet_create_conn(p, -1, bind_addr, INPORT_ANY, FALSE);
-  if (server_conn == NULL) {
-    int xerrno = errno;
+  xerrno = errno;
+
+  /* Restore the previous default inet family if necessary. */
+  if (bind_addr == NULL) {
+    (void) pr_inet_set_default_family(p, default_inet_family);
+  }
```

Comment:
This restoration of the default family for the Inet API is something that _should_ have been doing in the error case here; it's a learning from the actual bugfix in the Proxy FTP code.